### PR TITLE
History: remove OS specific workaround

### PR DESCRIPTION
**Pull request by telephon**

### CODE REVIEW

**Comment by brianlheim** [comment]
I'm doing some dev work on windows later today; I'll make sure to check this when I do. At a glance it looks OK. Thanks!

File: SCClassLibrary/Common/Streams/History.sc [hunk 1]
```diff
@@ -464,27 +480,17 @@ History { 		// adc 2006, Birmingham; rewrite 2007.
...
-			{
-				//
-				docTitle = title ++ Date.getDate.format("%Y-%m-%e-%Hh%M-History");
-				Document.new(docTitle, this.storyString)
-				// path not working yet
-				//	.path_(docTitle); // don't lose title.
-			}
-		)
+	document { |title = ""|
+		var docTitle = title ++ Date.getDate.format("%Y-%m-%e-%Hh%M-History");
```
Comment (brianlheim) [inline SCClassLibrary/Common/Streams/History.sc]: ahh, this is a problem. Windows' `strtime` function doesn't allow `%e` (day of month w/o leading zeroes). Two options for a fix (local fix anyway): use `%d` (day of month, _with_ leading zero), or use `%e` on unix and `%#d` on Windows (just tested, this already produces the correct result).

Perhaps later we should modify Date.format to internally convert %e to %#d on windows. As it stands, the documentation doesn't make this point clear (although it does make a note it depends on your system's strtime implementation).

**Comment by brianlheim** [comment]
forgot to note, it's all good aside from that. thanks!

**Comment by telephon** [comment]
thank you! shouldn't i perhaps make the change to Date.format now? The motivation for this PR is OS specific cleanup anyway. But anyway, I'm fine with either.

**Comment by brianlheim** [comment]
hmm.. I just reread the documentation for linux strftime. http://man7.org/linux/man-pages/man3/strftime.3.html

%e replaces the leading zero with a _space_, it doesn't just omit it. I think using %d would be better in this case, because it doesn't seem like this filename should have spaces.

I also haven't fully researched the difference between implementations on Windows, so my plan of attack would be to make a note of it in the docs for now and then open a separate ticket for the issue. Is that ok? The more I think about it the more I realize the fix might be more complicated than I thought.

**Comment by telephon** [comment]
So is it correct if I make a switch that uses `%d` on linux and `%#d` on windows? Does it have to be `%e` on macos (well I can test that myself).

**Comment by brianlheim** [comment]
Sorry, I should have been clearer about this. If the day is July 5, the outputs for old-school C strftime are as follows:

OS | %d | %e | %#d | reference
-- | -- | -- | -- | --
Windows | "05" | N/A | "5" (no padding) | https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man3/strftime.3.html
macOS | "05" | " 5" (leading space) | N/A | https://ss64.com/osx/syntax-strftime.html and https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man3/strftime.3.html
Linux | "05" | " 5" | N/A | http://man7.org/linux/man-pages/man3/strftime.3.html

I think switching it to `%d` for all platforms, and then leaving a note in the docs that this method exactly follows the behavior of these versions, is the reasonable solution. I don't think introducing a platform switch is worth the trouble for such a small detail.

**Comment by brianlheim** [comment]
@snappizz it is a bug fix actually! On Windows, `History` will now add timestamps where it previously used filler text.

**Pull request merged**